internal/injector/config/config-yml.go

```diff
@@ -0,0 +1,150 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package config
+
+import (
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/DataDog/orchestrion/internal/injector/aspect"
+	"gopkg.in/yaml.v3"
+)
+
+const OrchestrionYML = "orchestrion.yml"
+
+// loadYMLFile loads configuration from the specified directory.
+func (l *Loader) loadYMLFile(dir string, name string) (*configYML, error) {
+	filename := name
+	if !filepath.IsAbs(name) {
+		filename = filepath.Join(dir, name)
+	}
+	if !l.markLoaded(filename) {
+		// Already loaded, ignoring...
+		return nil, nil
+	}
+
+	yml, err := l.parseYMLFile(filename)
+	if err != nil {
+		return nil, err
+	}
+
+	dir = filepath.Dir(filename)
+	extends := make([]Config, 0, len(yml.Extends))
+	for _, ext := range yml.Extends {
+		extFilename := filepath.Join(dir, ext)
+
+		if stat, err := os.Stat(extFilename); err != nil {
+			return nil, maskErrNotExist(err)
+		} else if stat.IsDir() {
+			pkgs, err := l.packages(extFilename)
+			if err != nil {
+				return nil, fmt.Errorf("extends %q: %w", ext, err)
+			}
+			if len(pkgs) != 1 {
+				// This is not supposed to happen if `err == nil`.
+				panic(fmt.Errorf("extends %q: no package returned by packages.Load(%q)", ext, l.dir))
+			}
+
+			cfg, err := l.loadGoPackage(pkgs[0])
+			if err != nil {
+				return nil, maskErrNotExist(err)
+			}
+			if cfg == nil {
+				// Empty, nothing to do...
+				continue
+			}
```

Comment:
This case can't happen I believe but the case where everything inside `cfg` is empty is the one that is supposed to be covered. I will let you choose whenever you want to change caller or callee side